### PR TITLE
Bump backend dependencies and to latest stable rustc

### PR DESCRIPTION
**Pull request by jtgeibel**
r? @Turbo87

### CODE REVIEW

**Comment by jtgeibel** [comment]
I've resolved the merge conflict, but I think we'll want to deploy this separately from the rest of our backlog, so let's hold off on merging for now.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -91,8 +91,8 @@ jobs:
         run: |
           rustup set profile minimal
           # Pin to older version until a clippy regression is fixed
```
Comment (nipunn1313) [inline .github/workflows/ci.yml:93]: This comment seems to imply that it's desired to switch back to `rustup update stable` - which from git log history appears to be how it used to work.

Do you want to use
```
rustup update stable
rustup default stable
```


Alternately, remove this comment about pinning an older version?

**Comment by Turbo87** [comment]
@jtgeibel FYI I've rebased the branch. Should be ready to go from my side. Feel free to r=me.

**Comment by jtgeibel** [comment]
Thanks!

@bors r=@Turbo87

**Pull request merged**